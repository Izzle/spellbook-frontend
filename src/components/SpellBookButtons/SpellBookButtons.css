.SpellBookButtons__container {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}

.SpellBookButtons__button {
  background-color: blue;
  color: white;
  text-shadow: -1px -1px 0 #000,
      1px -1px 0 #000,
      -1px 1px 0 #000,
      1px 1px 0 #000;
  width: 150px;
  height: 40px;
  border: 1px solid #000;
  border-radius: 5px;
  margin-bottom: 15px;
}