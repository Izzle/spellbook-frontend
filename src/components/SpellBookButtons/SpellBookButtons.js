import React, { Component} from 'react';
import { Link } from 'react-router-dom';
import './SpellBookButtons.css';

export default class SpellBookButtons extends Component {

    // WE SHOULD DO CONDITIONAL RENDER
    // OR EVEN A HIGHER ORDER COMPONENT
    // LIKE IN THE ENDPOINTS SECTION
    // In one case, load the button that lets you EDIT your spellbook
    // in the other case, load the button that lets you SAVE your spellbook
    // ALSO. The 2nd button shouldnt do anything when you are building your spellbook- 
    //     so you don't accidentally click a link in the middle of your spellbook editing/building

    render() {
        return(<div>
            <button>edit spellbook</button>
            <Link to='/spellbook' className='SpellBookButtons__Link'>
                <button className='SpellBookButtons__button'>This button should tell you the name of the Spellbook you're in. Also, it will take you back to 'View Spellbook' page at /spellbook</button>
            </Link>
        </div>);
    }
}