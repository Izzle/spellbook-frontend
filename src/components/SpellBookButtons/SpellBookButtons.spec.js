import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import SpellBookButtons from './SpellBookButtons';
import { BrowserRouter } from 'react-router-dom';


describe('<SpellBookButtons />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter><SpellBookButtons /></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree = renderer.create(<BrowserRouter><SpellBookButtons /></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});