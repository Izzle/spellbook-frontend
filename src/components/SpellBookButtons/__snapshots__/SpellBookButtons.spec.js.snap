// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpellBookButtons /> renders this UI as expected 1`] = `
<div
  className="SpellBookButtons__container"
>
  <a
    className="SpellBookButtons__Link"
    href="/spellbook/undefined/build"
    onClick={[Function]}
  >
    <button
      className="SpellBookButtons__button"
    >
      Edit Spellbook
    </button>
  </a>
  <a
    className="SpellBookButtons__Link"
    href="/spellbook"
    onClick={[Function]}
  >
    <button
      className="SpellBookButtons__button"
    >
      View All Spellbooks
    </button>
  </a>
</div>
`;
